### PR TITLE
Android sample app

### PR DESCRIPTION
**Pull request by AniVerma17**
Create the sample Android app to demonstrate the new and updated features of ImageKit Android SDK (v3.0.1) and add it to the main collection.

### CODE REVIEW

File: android/README.md [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+
+```
+npm start
+```
+
+Node server will run at `http://localhost:8080`.
+
+
+> #### Recommendation
+> For ease for loading files, `app/src/main/java/io/imagekit/imagekitdemo/FileUtils.kt` can be imported into your project and used like it is demonstrated in `app/src/main/java/io/imagekit/imagekitdemo/UploadFileActivity.kt`.
```
Comment (ahnv) [inline android/README.md:81]:
```suggestion
> #### Recommendation
> For ease of loading files, `app/src/main/java/io/imagekit/imagekitdemo/FileUtils.kt` can be imported into your project and used like it is demonstrated in `app/src/main/java/io/imagekit/imagekitdemo/UploadFileActivity.kt`.
```

**Review by ahnv — CHANGES_REQUESTED** [review]

File: android/README.md (hunk 1 above)
Comment (AniVerma17) [reply android/README.md]: Done

File: android/README.md [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+Set the following keys in `.env`
+
+```shell
+PRIVATE_KEY=private_XXXXXXXXXXXX
+```
+
+API private key can be obtained from the [developer](https://imagekit.io/dashboard#developers) section in your ImageKit dashboard.
+
+> [!Note]
+> File upload will only work if, along with the above, you have also defined `publicKey`, `urlEndpoint`, and `authenticationEndpoint` variables in `app/config/Imagekit.js`.
```
Comment (ahnv) [inline android/README.md:63]: Please add the server files from https://github.com/imagekit-samples/quickstart/tree/master/ios/Server to android/server folder

Comment (AniVerma17) [reply android/README.md:63]: Done

File: android/app/src/main/java/io/imagekit/imagekitdemo/UploadFileActivity.kt [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
+        super.onActivityResult(requestCode, resultCode, data)
+        if (requestCode == 24 && resultCode == RESULT_OK) {
+            val selectedFile = data?.data
+            if (selectedFile != null){
+                file = FileUtils.getPath(this, selectedFile)?.let { File(it) }
+                uploadFile()
+            }
+        }
+    }
```
Comment (ahnv) [inline android/app/src/main/java/io/imagekit/imagekitdemo/UploadFileActivity.kt:47]: Getting an error while trying to upload a pdf file 

```
2024-06-03 11:11:03.384 16305-16305 AndroidRuntime          io.imagekit.imagekitdemo             E  
FATAL EXCEPTION: main
Process: io.imagekit.imagekitdemo, PID: 16305
java.lang.RuntimeException: Failure delivering result ResultInfo{who=null, request=24, result=-1, data=Intent { dat=content://com.android.providers.media.documents/... flg=0x1 }} to activity {io.imagekit.imagekitdemo/io.imagekit.imagekitdemo.UploadFileActivity}: java.lang.NullPointerException
	at android.app.ActivityThread.deliverResults(ActivityThread.java:5323)
	at android.app.ActivityThread.handleSendResult(ActivityThread.java:5362)
	at android.app.servertransaction.ActivityResultItem.execute(ActivityResultItem.java:67)
	at android.app.servertransaction.ActivityTransactionItem.execute(ActivityTransactionItem.java:45)
	at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2307)
	at android.os.Handler.dispatchMessage(Handler.java:106)
	at android.os.Looper.loopOnce(Looper.java:201)
	at android.os.Looper.loop(Looper.java:288)
	at android.app.ActivityThread.main(ActivityThread.java:7872)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:936)
Caused by: java.lang.NullPointerException
	at io.imagekit.imagekitdemo.FileUtils.getDataColumn(FileUtils.kt:144)
	at io.imagekit.imagekitdemo.FileUtils.getPath(FileUtils.kt:240)
	at io.imagekit.imagekitdemo.UploadFileActivity.onActivityResult(UploadFileActivity.kt:43)
	at android.app.Activity.dispatchActivityResult(Activity.java:8628)
	at android.app.ActivityThread.deliverResults(ActivityThread.java:5316)
	at android.app.ActivityThread.handleSendResult(ActivityThread.java:5362) 
	at android.app.servertransaction.ActivityResultItem.execute(ActivityResultItem.java:67) 
	at android.app.servertransaction.ActivityTransactionItem.execute(ActivityTransactionItem.java:45) 
	at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135) 
	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95) 
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2307) 
	at android.os.Handler.dispatchMessage(Handler.java:106) 
	at android.os.Looper.loopOnce(Looper.java:201) 
	at android.os.Looper.loop(Looper.java:288) 
	at android.app.ActivityThread.main(ActivityThread.java:7872) 
	at java.lang.reflect.Method.invoke(Native Method) 
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548) 
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:936) 
```

Comment (AniVerma17) [reply android/app/src/main/java/io/imagekit/imagekitdemo/UploadFileActivity.kt]: Fixed. Please review

**Review by Bloendodt — APPROVED** [review]

**Pull request merged**